### DISCUSSION TITLE
Is it possible to run integration tests with vitest?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [vitest-dev/vitest#6348] [Q&A]
I'm currently running a rather large integration test suite in Playwright in my project.
It's using playwrights ["Fixtures"](https://playwright.dev/docs/test-fixtures) feature to start a Sveltekit dev server, and connect to the correct in-memory database for each "worker". No browsers involved.

It works relatively well - however if I import certain functions from the Sveltekit app, playwright fails.

This made me wonder if Vitest is capable of running integration tests like this - or is vitest mostly for unit tests?

**Comment by sheremet-va** [comment]
Playwright always runs the code on the server. Vitest runs the code in the environment it was configured in, so when you write tests in browser mode for example, the test code runs in the browser.
